lib/dams/controller_helper.rb

```diff
@@ -44,6 +44,71 @@ def osf_contributors(document)
 		  osf_data = (osf_data.blank?) ? osf_data << {"name": "UC San Diego Library"} : osf_data
 		end
 
+		def osf_related_agents(document)
+			field_name = "relationship_json_tesim"
+			dams_data = document["#{field_name}"]
+			osf_data =[]
+
+			if dams_data != nil
+		    dams_data.each do |datum|
+		    	
+		      relationships = JSON.parse(datum)
+		    	relationships.each do |key, value|
+		    		value.each do |v|
+		   				osf_data << {agent_type: agent_type(key), type: "Person", "name": v}
+		   			end
+		   		end
+		    end
+		  end
+		  osf_data << {agent_type: "Publisher", type: "Organization", "name": "UC San Diego Library Digital Collections"}
+		end
+
+		def agent_type(type)
+			type = 'principalinvestigator' if type == 'principal investigator' || type == 'Principal Investigator'
+
+			share_agent_type = [
+				'AGENTWORKRELATION',
+				'AgentWorkRelation',
+				'CONTRIBUTOR',
+        'CREATOR',
+        'Contributor',
+        'Creator',
+        'FUNDER',
+        'Funder',
+        'HOST',
+        'Host',
+        'PRINCIPALINVESTIGATOR',
+        'PRINCIPALINVESTIGATORCONTACT',
+        'PUBLISHER',
+        'PrincipalInvestigator',
+        'PrincipalInvestigatorContact',
+        'Publisher',
+        'agentworkrelation',
+        'contributor',
+        'creator',
+        'funder',
+        'host',
+        'principalinvestigator',
+        'principalinvestigatorcontact',
+        'publisher'
+       ]
+      type = (share_agent_type.include? type) ? type : 'Contributor'
+		end
+
+		def osf_extra(document)
+      field_name = 'otherNote_json_tesim'
+      dams_data = document["#{field_name}"]
+      osf_data = {}
+
+			if dams_data != nil
```

Comment:
Could there perhaps be a guard clause here? Something like `return osf_data unless dams_data`?